pysmarthashtag/api/authentication.py

```
@@ -168,99 +171,244 @@ async def _refresh_access_token(self):
+                )
+                raise SmartAPIError(f"Gigya login error (code={gigya_code}): {gigya_message}")
+
+            try:
+                session_info = login_result["sessionInfo"]
+                login_token = session_info["login_token"]
                 expires_at = datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(
-                    seconds=int(login_result["sessionInfo"]["expires_in"])
+                    seconds=int(session_info["expires_in"])
                 )
             except (KeyError, ValueError):
                 raise SmartAPIError("Could not get login token from login page")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)